frontend/src/components/pages/FridgeManagement/FridgeCheckIns/DeleteCheckIns.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  HStack,
+  Input,
+  Text,
+  useDisclosure,
+  useToast,
+} from "@chakra-ui/react";
+import { endOfDay } from "date-fns";
+import React, { useState } from "react";
+import DatePicker, { DateObject } from "react-multi-date-picker";
+import { useHistory } from "react-router-dom";
+
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import DeleteShiftModal from "./DeleteShiftModal";
+import ErrorMessages from "./ErrorMessages";
+
+const DeleteCheckInsPage = () => {
+  const toast = useToast();
+  const history = useHistory();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const [dateRange, setDateRange] = useState<DateObject[]>([
+    new DateObject(),
+    new DateObject().add(1, "days"),
+  ]);
+  const [dateRangeError, setDateRangeError] = useState("");
+
+  const validateForm = () => {
+    let newError = "";
+    let valid = true;
+    if (!dateRange[0] || !dateRange[1]) {
+      valid = false;
+      newError = ErrorMessages.bothDateFieldsRequired;
+    }
+    setDateRangeError(newError);
+    return valid;
+  };
+
+  const openModal = () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    onOpen();
+  };
+  const onDeleteClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+    const res = await CheckInAPIClient.deleteCheckInsByDateRange(
+      dateRange[0].toString(),
```

Comment:
this will always be `startOfDay` from the calendar picker?

Reply:
the calendar picker uses the current time

Reply:
when i consoled.log the toString(), it was just the date, e.g. 10/04/2022, and when it went to backend it became 12 am UTC that day, i think we can add startOfDay() here so that when it goes to backend it is 3:59 am (a bit more accurate, and the donations that are scheduled right now after 8pm EST, they are deleted for a day before. So if i delete from apr 9 - 11, the apr 8 one after 8pm will also be deleted, so startOfDay should fix that.